sdk/log/example_test.go

```diff
@@ -112,8 +112,57 @@ func ignoreLogs(ctx context.Context) bool {
 	return ok
 }
 
+// Use a processor which sets EventName on log records having "event.name" string attribute.
+// This is useful for users of logging libraries that do not support
+// setting the event name on log records, but do support attributes.
+func ExampleProcessor_eventName() {
+	// Existing processor that emits telemetry.
+	var processor log.Processor = log.NewBatchProcessor(nil)
+
+	// Add a processor so that it sets EventName on log records.
+	eventNameProcessor := &EventNameProcessor{}
+
+	// The created processor can then be registered with
+	// the OpenTelemetry Logs SDK using the WithProcessor option.
+	_ = log.NewLoggerProvider(
+		// Order is important here. Redact before handing to the processor.
+		log.WithProcessor(eventNameProcessor),
+		log.WithProcessor(processor),
+	)
+}
+
+// EventNameProcessor is a [log.Processor] that sets the EventName
+// on log records that have an attribute with a key containing "event.name"
+// and a value of type string.
+// It is useful for logging libraries that do not support
+// setting the event name on log records,
+// but do support attributes.
+type EventNameProcessor struct{}
+
+// OnEmit sets the EventName on log records that have an attribute
+// with a key containing "event.name" and a value of type string.
+func (p *EventNameProcessor) OnEmit(ctx context.Context, record *log.Record) error {
+	record.WalkAttributes(func(kv logapi.KeyValue) bool {
+		if strings.Contains(strings.ToLower(kv.Key), "event.name") && kv.Value.Kind() == logapi.KindString {
```

Comment:
Is there no const for `event.name` in the semconv package? Should there be? Actually, I see that the event attribute is deprecated: https://opentelemetry.io/docs/specs/semconv/registry/attributes/event/.

Maybe a gap in my understanding, but I see that EventName is PascalCase here: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.47.0/specification/logs/data-model.md#field-eventname. What is the difference between `event.name` and `EventName` in this context?

Reply:
Actually, I think this better be  be a new `otel.event_name` attribute and not `event.name`.
Based on https://github.com/open-telemetry/semantic-conventions/blob/main/docs/registry/attributes/otel.md#otel-scope-attributes.

I created:
- https://github.com/open-telemetry/semantic-conventions/issues/2597